### PR TITLE
Add 'All Offices' custom logic.

### PR DESCRIPTION
**Pull request by glogiotatidis**

### CODE REVIEW

File: careers/careers/static/js/filters.js [hunk 1]
```diff
@@ -114,9 +114,30 @@
...
+
+                if (data.indexOf(value + ',') === -1) {
+                    element.classList.add('hidden');
+                }
+            });
+        },
+        filterLocations: function(value) {
+            this.$positionTable.find('tr.position').each(function(index, element) {
+                var data = element.dataset.location;
+                if (!value)
```
Comment (jpetto) [inline careers/careers/static/js/filters.js]: Could this check happen prior to the `.each` loop to save some cycles?

File: careers/careers/static/js/filters.js [hunk 2]
```diff
@@ -114,9 +114,30 @@
          */
         filterPositions: function(field, value) {
             this.$positionTable.find('tr.position').each(function(index, element) {
+                var data = element.dataset[field];
                 if (!value)
```
Comment (jpetto) [inline careers/careers/static/js/filters.js]: Same here - check for no value before looping?

File: careers/careers/static/js/filters.js [hunk 3]
```diff
@@ -114,9 +114,30 @@
...
+            this.$positionTable.find('tr.position').each(function(index, element) {
+                var data = element.dataset.location;
+                if (!value)
+                    return;
+
+                // When user selects 'Remote' only list jobs explicitly marked
+                // Remote otherwise list jobs matching value (which is a mozilla
+                // office) and those marked as 'All Offices'
+                if (value === 'Remote') {
+                    if (data.indexOf(value + ',') === -1) {
```
Comment (jpetto) [inline careers/careers/static/js/filters.js]: Could these be combined into a single `if`?

Comment (glogiotatidis) [inline careers/careers/static/js/filters.js]: If combined and `value === Remote` but not in `data` then `else if` will be checked which is not the desired functionality.

File: careers/careers/static/js/filters.js (hunk 2 above)
Comment (glogiotatidis) [inline careers/careers/static/js/filters.js]: 👍

File: careers/careers/static/js/filters.js (hunk 1 above)
Comment (glogiotatidis) [inline careers/careers/static/js/filters.js]: 👍

File: careers/careers/static/js/filters.js (hunk 3 above)
Comment (jpetto) [inline careers/careers/static/js/filters.js]: Ah, makes sense. Carry on.

**Comment by jpetto** [comment]
Couple small suggestions, but otherwise looks good. Didn't get the thing running locally, so I assume functionality is as expected. 😁

**Comment by glogiotatidis** [comment]
Thanks for your time @jpetto

**Pull request merged**